src/server/saved_objects/client/saved_objects_client.js

```diff
@@ -175,26 +178,62 @@ export class SavedObjectsClient {
       return { saved_objects: [] };
     }
 
-    const docs = objects.map(doc => {
-      return { _type: get(doc, 'type'), _id: get(doc, 'id') };
-    });
+    const docs = objects.reduce((acc, { type, id }) => {
+      acc.push({}, createIdQuery({ type, id }));
+      return acc;
+    }, []);
+
 
-    const response = await this._withKibanaIndex('mget', { body: { docs } })
-      .then(resp => get(resp, 'docs', []).filter(resp => resp.found));
+    const response = await this._withKibanaIndex('msearch', { body: docs })
+    .then(resp => {
+      let results = [];
+      const responses = get(resp, 'responses', []);
+      responses.forEach(r => {
+        results = results.concat(get(r, 'hits.hits'));
+      });
+      return results;
+    });
 
     return {
       saved_objects: response.map(r => {
+        const docType =  getDocType(r);
+
         return {
           id: r._id,
-          type: r._type,
+          type: docType,
           version: r._version,
-          attributes: r._source
+          attributes: get(r, `_source.${docType}`) || r._source
         };
       })
     };
   }
 
   /**
+   * Gets a single object
+   *
+   * @param {string} type
+   * @param {string} id
+   * @returns {promise} - { id, type, version, attributes }
+   */
+  async get(type, id) {
+    const response = await this._withKibanaIndex('search', { body: createIdQuery({ type, id }) });
+    const [hit,] = get(response, 'hits.hits', []);
```

Comment:
Thought it was needed to ignore the additional assignment, but appears it's not necessary. Will remove.